cdap-docs/admin-manual/source/operations/tx-maintenance.rst

```diff
@@ -55,3 +58,42 @@ with the :ref:`Transaction Service HTTP RESTful API <http-restful-api-transactio
 The current length of the invalid transaction list can be obtained using 
 :ref:`a call <http-restful-api-transactions-number>` 
 with the :ref:`Transaction Service HTTP RESTful API <http-restful-api-transactions-number>`.
+
+
+Using the Queue Debugger Tool
+=============================
+The Queue Debugger Tool allows you to look at the entries in queues, and can be useful in
+solving problems with transactions. This is a debug tool for a queue, returning
+information such as how many entries are in a queue, how many have been processed, and how
+many are in progress. 
+
+Background
+----------
+Each flow has a queue table. Within each queue table are multiple queues. There is one
+queue for each connection between flowlets. As each flowlet may have multiple consuming
+flowlets, this results in there being multiple queues.
+
+The name of a queue is determined by the producer flowlet. The consumer flowlet (required
+for the command line parameters of the tool) is referred to as the consumer. 
+
+Running the Tool
+----------------
+It's important that the tool be run with the same classpath as CDAP Master to avoid
+problems with the ordering of classes and to ensure that the CDAP classes appear before
+the HBase classpath. This is to avoid the invocation of any older versions of the ASM
+library that are present in the HBase classpath.
+
+The easiest way to start the tool with the same classpath as CDAP Master is to use::
+
+  $ /etc/init.d/cdap-master run co.cask.cdap.data.tools.HBaseQueueDebugger
+  
+or::
+
+  $ /opt/cdap/master/bin/svc-master run co.cask.cdap.data.tools.HBaseQueueDebugger
+  
+Running the ``help`` option will give a summary of commands and required parameters.
+
+The tool ``SimpleHBaseQueueDebugger`` is a wrapper of the tool that uses common defaults. 
```

Comment:
`that uses a set of defaults useful for displaying the minimum transaction time for all events in all queues.`